DuolearnHQ/Tasky
# Feat/jwt service

##  #3 Title: Refactor JWT Token Handling and User Creation Service

### Description:
This pull request introduces significant refactoring to improve the architecture and maintainability of our authentication and user management modules. Following the SOLID principles, specifically the Single Responsibility Principle, I have decoupled the JWT token creation and validation from the user creation service. Additionally, to enhance the validation of data in our routes, I have integrated the Zod package.

### Key Changes:

#### Separate JWT Service:

<ul>
<li>Created a new JwtService dedicated to handling JWT token creation and validation.</li>
<li>This service manages all aspects of JWT, including token generation, expiration, and verification.</li>
<ul>

#### Refactoring User Creation Service:

<ul>
<li>Removed JWT token logic from the UserCreationService.</li>
<li>Now, the service strictly adheres to its core responsibility - creating and managing user data.</li>
<li>This change makes the service more focused, easier to maintain, and aligned with the Single Responsibility Principle.</li>
</ul>

#### Enhanced Data Validation with Zod:

<ul>
<li>Implemented Zod for robust schema validation in our routes.</li>
<li>Replaced previous inline validation logic with structured Zod schemas.</li>
<li>This enhancement ensures that all incoming data is correctly formatted and valid before processing.</li>
</ul>

Motivation:
The primary motivation behind these changes is to align our codebase more closely with the SOLID principles. By ensuring that each module or service has a single responsibility, we can create a more scalable, maintainable, and testable application. The introduction of Zod for validation further standardizes our approach to handling incoming data and reduces the potential for errors.

### How to Test:
<ol>
<li>Pull the branch associated with this PR.</li>
<li>Perform manual testing to verify JWT functionality and user creation workflows.</li>
</ol>

### Request for Feedback:
#### I would appreciate feedback on:
<ul>
<li>The structure and clarity of the new JwtService.</li>
<li>The integration of Zod and its schema designs.</li>
<li>Any potential edge cases or error handling scenarios I might have missed.</li>
</ul>
@pavankpdev
